pkg/arvo/sys/zuse.hoon

```diff
@@ -5085,18 +5082,41 @@
 ::  WARNING: ordered-map will not work properly if two keys can be
 ::  unequal under noun equality but equal via the compare gate
 ::
-++  ordered-map
+++  on
+  ~/  %on
   |*  [key=mold val=mold]
   =>  |%
-      +$  item  (mk-item key val)
+      +$  item  [key=key val=val]
       --
   ::  +compare: item comparator for horizontal order
   ::
+  ~%  %comp  +>+  ~
   |=  compare=$-([key key] ?)
+  ~%  %core    +  ~
   |%
-  ::  +check-balance: verify horizontal and vertical orderings
+  ::  +all: apply logical AND boolean test on all values
+  ::
+  ++  all
+    ~/  %all
+    |=  [a=(tree item) b=$-(item ?)]
+    ^-  ?
+    |-
+    ?~  a
+      &
+    ?&((b n.a) $(a l.a) $(a r.a))
+  ::  +any: apply logical OR boolean test on all values
+  ::
+  ++  any
+    ~/  %any
+    |=  [a=(tree item) b=$-(item ?)]
+    |-  ^-  ?
+    ?~  a
+      |
+    ?|((b n.a) $(a l.a) $(a r.a))
+  ::  +apt: verify horizontal and vertical orderings
   ::
-  ++  check-balance
+  ++  apt
+    ~/  %apt
     =|  [l=(unit key) r=(unit key)]
```

Comment:
This hint is wrong. `~/  %foo` is `~%  %foo  +>  ~`, but, in this case, the parent core is at `+>+`, not `+>`.

You could fix the hint, or move `l` and `r` inside the gate and add an inner `|-`. That's probably the right thing to do for jetting anyway, since`l` and `r` should be controlled by the caller.

Reply:
Fixed, thanks.